charts/newrelic-logging/templates/configmap.yaml

```diff
@@ -5,66 +5,28 @@ metadata:
   labels: {{ include "newrelic-logging.labels" . | indent 4 }}
   name: {{ template "newrelic-logging.fluentBitConfig" . }}
 data:
-  # Configuration files: server, input, filters and output
-  # ======================================================
+  {{- if (include "newrelic-logging.lowDataMode" .) }}
+  {{ $lowDataDefault := .Files.Get "static/lowdatamodedefaults.yaml" }}
+    {{- $lowDataDefault }}
+  {{- else }}
   fluent-bit.conf: |
-    [SERVICE]
-        Flush         1
-        Log_Level     ${LOG_LEVEL}
-        Daemon        off
-        Parsers_File  parsers.conf
-        HTTP_Server   On
-        HTTP_Listen   0.0.0.0
-        HTTP_Port     2020
-
-    [INPUT]
-        Name              tail
-        Tag               kube.*
-        Path              ${PATH}
-        Parser            ${LOG_PARSER}
-        DB                ${FB_DB}
-        Mem_Buf_Limit     7MB
-        Skip_Long_Lines   On
-        Refresh_Interval  10
-
-    [FILTER]
-        Name           kubernetes
-        Match          kube.*
-        # We need the full DNS suffix as Windows only supports resolving names with this suffix
-        # See: https://kubernetes.io/docs/setup/production-environment/windows/intro-windows-in-kubernetes/#dns-limitations
-        Kube_URL       https://kubernetes.default.svc.cluster.local:443
-        K8S-Logging.Exclude ${K8S_LOGGING_EXCLUDE}
-{{- if (include "newrelic.lowDataMode" .) }}
-        Labels         Off
-        Annotations    Off
-
-    [FILTER]
-        Name           nest
-        Match          *
-        Operation      lift
-        Nested_under   kubernetes
-{{- end }}
-
-    [FILTER]
-        Name           record_modifier
-        Match          *
-        Record         cluster_name ${CLUSTER_NAME}
-{{- if (include "newrelic.lowDataMode" .) }}
-        Allowlist_key  container_name
-        Allowlist_key  namespace_name
-        Allowlist_key  pod_name
-        Allowlist_key  stream
-        Allowlist_key  log
-{{- end }}
-
-    [OUTPUT]
-        Name           newrelic
-        Match          *
-        licenseKey     ${LICENSE_KEY}
-        endpoint       ${ENDPOINT}
-        lowDataMode    ${LOW_DATA_MODE}
-
+    {{- if .Values.fluentBit.config.service }}
+      {{- .Values.fluentBit.config.service | nindent 4 }}
+    {{ end }}
+    {{- if .Values.fluentBit.config.inputs }}
+      {{- .Values.fluentBit.config.inputs | nindent 4 }}
+    {{ end }}
+    {{- if .Values.fluentBit.config.filters }}
+      {{- .Values.fluentBit.config.filters | nindent 4 }}
+    {{ end }}
```

Comment:
So here, we would have (using pseudocode, I didn't check the syntax):

```
    {{- if .Values.fluentBit.config.filters AND !lowDataMode }}
      {{- .Values.fluentBit.config.filters | nindent 4 }}
    {{ end }}
    {{- if .Values.fluentBit.config.lowDataModeFilters AND lowDataMode }}
      {{- .Values.fluentBit.config.lowDataModeFilters | nindent 4 }}
    {{ end }}
```

This way, we avoid having to duplicate (and maintain) a complete file for `lowDataMode` alone.